### DISCUSSION TITLE
Met data needed for correct crop yield simulation in APSIM

### DISCUSSION THREAD

**Discussion by ankanasadhu** [APSIMInitiative/ApsimX#9183] [Q&A]
What is the duration of met data required for correct crop yield simulation in APSIM?
Let us say that the harvest date is October 16 then will I get the correct simulation for met data inputs until October 1?

**Comment by par456** [comment]
Apsim requires a met file with weather for every day that is simulated, so if your harvest date is October 16th and your met file ends at October 1st, you'll get an error and won't reach harvest.

But if you fill your met file with dummy weather values for the days after October 1st, it will react based as if that was the real weather. How that will effect the crop will differ depending which crop is planted and how much 'made up' weather you've had to include.

Is there a reason your met file ends before harvest? is there some other way to find those missing days from another reliable source?

**Reply by ankanasadhu** [reply]
I am building a hybrid model (Physics Informed ML model) for prediction of yield. This model must predict crop yield 15 days ahead of harvest. I am thinking of using previous year's met data for those 15 days. Any suggestions?

**Comment by ankanasadhu** [comment]
I am building a hybrid model (Physics Informed ML model) for prediction of yield. This model must predict crop yield 15 days ahead of harvest. I am thinking of using previous year's met data for those 15 days. Any suggestions?

**Reply by LenLon** [reply]
Don't you need the Apsim-simulated yields to evaluate the ML-predicted yields against anyway?

**Reply by ankanasadhu** [reply]
Yes I do.

**Reply by LenLon** [reply]
Then you simulate the yields using the full .met files with ApsimX, and then train the ML model on the .met files with 15 days cut off from the end as well as the ApsimX simulated yields. The ML model does the forecasting, and the quality of the forecast is evaluated against the simulated ApsimX yields (the "real" yield).
